doc/design/vic-machine.md

```diff
@@ -0,0 +1,180 @@
+It is not intended that there be RBAC within a VCH at this time.
+
+How this multi-tenacy model is used is left up to the business, but there are two primary models that we consider during development:
+* team based - the VCH is assigned to a team for their use, potentially running a mix of independent workloads
+* application based - a VCH is used as the management construct for a given appliancation, with all containers being portions of that app
+
+
+## VCH manifest
+
+Certain information is required to deploy a VCH, with the _viadmin_ and _admin_ each contributing portions of that data. The VCH manifest is the mechanism by which that cooperation happens, and is the token via which authority delegation from _viadmin_ to _admin_ occurs. Creation of a manifest is conceptually a compositing process with the following inputs and an immutable result:
+
+1. input manifest (if omitted, this creates a new manifest from scratch)
+2. a restriction set of some kind ( e.g. compute resource path, datastore prefix, registry whitelist)
+  - as a note, user credentials are also considered a restriction as they control permissible operations
+
+
+The _viadmin_ uses vic-machine to create a _base manifest_ containing the following restrictions:
+* target vSphere environment
+* vSphere user and credential for created VCHs:
+  - must already exist, or
+  - must be created by vic-machine during manifest creation, or
+  - demand created during VCH deployment, requiring stored admin credentials:
+* stored _viadmin_ credentials if necessary:
+    - encrypted credentials and [validating proxy](components.md#validating-proxy) URI, or
+    - unencrypted credentials (_viadmin_ and _admin_ roles are held by the same entity and manifests are stored securely)
+
+The _base manifest_ is the minimum set of informtion necessary from the _viadmin_ role. There is additional information that will almost always be required for full function of a VCH:
+* vSphere network to use for container network
+  - must already exist
+* vSphere switch to use for container network:
+  - must already exist
+  - if switch, port group or network is demand created during VCH deployment, requiring stored admin credentials
+    - in an enterprise environment, creation of a portgroup without VXLAN or VLAN is unlikely to suffice, so pre-existing portgroup is highly recommended
+
+To create a VCH, disambiguation of which resource to use from the available set is necessary. If there is no ambiguity, i.e only one of each, then this can be omitted:
+* _client_ network - necessary for any access to the VCH
+* default container network:
+  - _private_ port group or dedicated vSwitch, or
+  - IPAM config for _client_ network (defaults to DHCP)
+* datastore paths for (may all be the same path):
+  - images
+  - containers
+  - volumes
+
+
+## Implementation approach
+
+My current approach is to have several components that get rolled into both vic-machine. Initial implementation should focus on:
```

Comment:
This was actually intentional - this doc is one of the ways I'm communicating with Emma and I wanted a less definite phrase that "The current approach".
I've altered it to "Recommended initial approach" ...